docs/user-guide/recipes.rst

```diff
@@ -96,88 +96,50 @@ to which that consumer is bound, run the following command.
+r10k to connect to Pulp instead of the default puppet forge. See the r10k
+documentation for details and the specific location of this file in your environment.
 
-For technical reasons described in the note below, the ``puppet module install``
-tool in versions prior to 3.3 ignores the part of the URL after the host name,
-which means we cannot put the repository ID in the URL. We have a work-around
-that will still allow you to use the ``puppet module install`` command with Pulp,
+For technical reasons described in the note below, r10k ignores the part of the
+URL after the host name, which means we cannot put the repository ID in the URL.
+We have a work-around that will still allow you to use r10k with Pulp,
 and it involves the use of basic auth credentials as part of the URL.
 
-.. note:: Puppet Forge implements a web API that their client uses to obtain dependency
-          data when installing a module. Unfortunately, their command line tool has
-          hard-coded absolute paths instead of relative, which means the API must live at
-          the root of a web server. As a result, we cannot put the repository ID in the
-          path as you would expect with the above example.
+.. note:: Puppet Forge implements a web API that r10k uses to obtain the download
+          location of modules that it needs to install. Unfortunately, r10k has
+          hard-coded absolute paths instead of relative, which means the API must
+          live at the root of a web server. As a result, we cannot put the repository
+          ID in the path as you would expect with the above example.
 
 - **Consumer ID** For a consumer registered with Pulp, just specify its consumer
   ID as the username in the URL, and a "." for the password. A consumer's ID is a
   unique identifier just like a username, so this isn't actually a bad use of
   that field. When a consumer ID is provided, Pulp searches all of that consumer's
-  bound repositories for either the newest version of the requested module, or
-  if a version is specified, searches for the exact version requested. Once a
-  suitable module has been located in a bound repository, all dependency data
-  returned is scoped to that same repository.
+  bound repositories for the exact version of each module defined in the Puppetfile.
 
 ::
 
-  $ puppet module install --module_repository http://consumer1:.@localhost
+  forge:
+    baseurl: 'http://consumer1:.@localhost' 
 
 - **Repository ID** For machines that are not bound to a repository, or for a
   bound machine where you want to specify a repository, do so in the password
   field. If a repository ID is specified, any value in the username field is
   ignored. To keep the convention, use a single "." as a null value.
 
-::
-
-  $ puppet module install --module_repository http://.:forge@localhost
-
-The repository URL can be set in ``/etc/puppet/puppet.conf`` so that it
```

Comment:
This part is still true. Maybe we should keep this somewhere?

Reply:
Let me test/double check and let you know. According to the docs, that was specific to pre-Puppet 3.3.x (`r10k` and `puppet module` implement API calls differently)

Reply:
@ekohl It appears that the `Puppet Client < 3.3` is the only accurate way of installing with the CLI tool now. I'm not sure if something in the Puppet Agent changed that broke the process described for post-3.3 installations or if it was always incorrect, but I have fixed that in the PR.

Reply:
Have you tried with the setting in `[main]` and `[agent]`? Do Either of them work? https://puppet.com/docs/puppet/5.4/configuration.html#modulerepository is still documented as an option.

Reply:
@ekohl The `puppet module install --module_repository=http://localhost/pulp_puppet/forge/repository/demo puppetlabs-stdlib` doesn't work at all (at least with Puppet 5.4 and 5.5) and only the "legacy" format works in modern puppet